remove Dockerfile.e2e

crazy-max opened the pull request:
`Dockerfile.e2e` is not used anymore. Integration tests run through the main `Dockerfile`.

Also remove the daemon OS/Arch detection script that is not necessary anymore. It was used to select the `Dockerfile` based on the arch like `Dockerfile.arm64` but we don't have those anymore. Was also used to check referenced frozen images in the `Dockerfile` or `Dockerfile.e2e`.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

crazy-max commented on hack/make/build-integration-test-binary:
```diff
@@ -1,7 +0,0 @@
-#!/usr/bin/env bash
-# required by https://github.com/AkihiroSuda/kube-moby-integration
```
@AkihiroSuda https://github.com/AkihiroSuda/kube-moby-integration is archived. LGTY to remove this file?

AkihiroSuda replied:
LGTY

AkihiroSuda commented on testutil/fixtures/load/frozen.go:
```diff
@@ -111,11 +111,7 @@ func pullImages(ctx context.Context, client client.APIClient, images []string) e
-		dockerfile = "Dockerfile"
-	}
-	dockerfilePath := filepath.Join(filepath.Dir(filepath.Clean(cwd)), dockerfile)
+	dockerfilePath := filepath.Join(filepath.Dir(filepath.Clean(cwd)), "Dockerfile")
```
This part might be safe to retain.
Somebody might be potentially using it?

crazy-max committed:
remove Dockerfile.e2e

Dockerfile.e2e is not used anymore. Integration tests run
through the main Dockerfile.

Also removes the daemon OS/Arch detection script that is not
necessary anymore. It was used to select the Dockerfile based
on the arch like Dockerfile.arm64 but we don't have those
anymore. Was also used to check referenced frozen images
in the Dockerfile.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

tianon commented on hack/make/.integration-daemon-start:
```diff
@@ -149,3 +149,5 @@ while ! ${TEST_CLIENT_BINARY} version &> /dev/null; do
 done
 printf "\n"
+
+bundle .ensure-emptyfs
```
Does this not need to still exclude Windows?

tianon replied:
If so, it could just be a simplified version of that detection script like `[ "$(docker version --format '{{ .Server.Os }}')" != 'windows' ]` (especially now that we can probably pretty reasonably assume `docker version -f` support :joy:)

AkihiroSuda commented:
Needs rebase
